src/utils/io/random_access.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use positioned_io::{ReadAt, Size};
+use std::fs::File;
+use std::io;
+use std::path::Path;
+
+/// This wrapper is needed, because [`positioned_io::RandomAccessFile`] does not support
+/// [`positioned_io::Size`].
```

Comment:
Open a PR with positioned io and add a tracking issue?

Reply:
Yes, this will happen soon. Unfortunately it does not look like they accept PRs or maintain the repository.

Reply:
https://github.com/ChainSafe/forest/issues/3341